cmd/create/cluster/cmd.go

```diff
@@ -1917,6 +1943,113 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	isSharedVPC := false
+	privateHostedZoneID := strings.Trim(args.privateHostedZoneID, " \t")
+	privateHostedZoneRoleARN := strings.Trim(args.privateHostedZoneRoleARN, " \t")
+	baseDomain := strings.Trim(args.baseDomain, " \t")
+	if privateHostedZoneID != "" ||
+		privateHostedZoneRoleARN != "" ||
+		baseDomain != "" {
```

Comment:
Currently, the way we identify a shared VPC in the backend is:
```
func (c *Cluster) IsSharedVpc() bool {
	return c.AWS != nil && (c.AWS.PrivateHostedZoneID != nil || c.AWS.PrivateHostedZoneRoleARN != nil)
}
```
It is true that `baseDomain` is required as well, but there is another effort to enable the user to provide `baseDomain` outside of the scope of the shared VPC. So I think it's ok to check only `privateHostedZoneID` or `privateHostedZoneRoleARN`.

Reply:
updated logic here